platform/pulpcore/app/viewsets/repository.py

```diff
@@ -114,6 +114,18 @@ class ImporterViewSet(NamedModelViewSet):
     endpoint_name = 'importers'
     filter_class = ImporterFilter
 
+    def update(self, request, *args, **kwargs):
+        partial = kwargs.pop('partial', True)
+        importer = self.get_object()
+        serializer = self.get_serializer(importer, data=request.data, partial=partial)
+        serializer.is_valid(raise_exception=True)
+        async_result = tasks.importer.update.apply_async_with_reservation(
+            tags.RESOURCE_REPOSITORY_TYPE, importer.repository.name,
+            kwargs={'importer_id': importer.id,
```

Comment:
apply_async_with_reservation has a args keyword argument which is for the position arguments to pass on to the task. I think that's what should be used here, since that is what update() is expecting.

```
async_result = tasks.importer.update.apply_async_with_reservation(
             tags.RESOURCE_REPOSITORY_TYPE, importer.repository.name,
             args=(importer.id, ), kwargs={'data': request.data} )
```